### PR TITLE
Updated sim, added base classes, made less dumb player

### PR DESCRIPTION
**Pull request by adamreis**
Does this all make sense?  Take a look at the Location/Post classes.  I wasn't able to actually start on clustering, because it took so long to figure out how everything was working, but at least things are structured better now.

### CODE REVIEW

File: src/outpost/dumb/Player.java [hunk 1]
```diff
@@ -79,12 +79,12 @@ public int delete(ArrayList<ArrayList<Pair>> king_outpostlist, Point[] gridin) {
...
-    	if (prarr.size()>noutpost) {
-			movePair mpr = new movePair(prarr.size()-1, new Pair(0,0), true);
+    	/*if (prarr.size()>noutpost) {
+			movePair mpr = new movePair(prarr.size()-1, new Pair(0,0));
 			nextlist.add(mpr);
 			//mpr.printmovePair();
-		}
-    	else {
+		}*/
+    	//else {
```
Comment (kevin-roark) [inline src/outpost/dumb/Player.java:87]: why all these comments instead of deletions?

File: src/outpost/group1/Player.java [hunk 2]
```diff
@@ -8,25 +8,27 @@
...
 
-		private Point[] grid;
-		private int[] theta = new int[100];
-
-		private int turn;
-
 		public Player(int id) {
 			super(id);
+			switch (id) {
+				case 0: this.baseLoc = new Location(0,0);
```
Comment (kevin-roark) [inline src/outpost/group1/Player.java:19]: does the simulator not rotate things automatically?

File: src/outpost/sim/Outpost.java [hunk 3]
```diff
@@ -700,10 +720,15 @@ void playStep() {
...
+				if (king_outpostlist.get(d).size()>noutpost[d]) {
+					int removedid = players[d].delete(king_outpostlist, grid);
+					king_outpostlist.get(d).remove(removedid);
+					System.out.printf("player %d delete outpost %d\n", d, removedid);
+				}
+				nextlist = players[d].move(king_outpostlist, grid);
 				for (int i=0; i<nextlist.size(); i++) {
-					movePair next = new movePair();
-					next = nextlist.get(i);
+					//movePair next = new movePair();
```
Comment (kevin-roark) [inline src/outpost/sim/Outpost.java:730]: y

**Comment by kevin-roark** [comment]
lgtm sans comments breH

File: src/outpost/dumb/Player.java (hunk 1 above)
Comment (adamreis) [inline src/outpost/dumb/Player.java:87]: No clue, this is all stuff that Bingyi added

File: src/outpost/sim/Outpost.java (hunk 3 above)
Comment (adamreis) [inline src/outpost/sim/Outpost.java:730]: No clue, this is all stuff that Bingyi added

File: src/outpost/group1/Player.java (hunk 2 above)
Comment (adamreis) [inline src/outpost/group1/Player.java:19]: Of course not, breh.  That'd be too easy.

Comment (kevin-roark) [inline src/outpost/group1/Player.java:19]: lol awesome

**Pull request merged**